### ISSUE TITLE
UnicodeDecodeError parsing PKG-INFO

### ISSUE THREAD

**Issue by jaraco** [pypa/setuptools-scm#28]
On Python 3.4.2 in some environments (maybe when LANG=C), setuptools_scm raises a UnicodeDecodeError when it encounters non-ascii in the PKG-INFO file:

```
wget https://pypi.python.org/packages/source/i/irc/irc-12.2.tar.gz#md5=efbb4d9a6bb92b24ab6109e04716ef38 -O - | tar xz
cd irc-12.2
python
>>> import setuptools_scm       
>>> setuptools_scm.get_version()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/app/env/lib/python3.4/site-packages/setuptools_scm-1.5.2-py3.4.egg/setuptools_scm/__init__.py", line 43, in get_version
  File "/app/env/lib/python3.4/site-packages/setuptools_scm-1.5.2-py3.4.egg/setuptools_scm/__init__.py", line 21, in version_from_scm
  File "/app/env/lib/python3.4/site-packages/setuptools_scm-1.5.2-py3.4.egg/setuptools_scm/hacks.py", line 8, in parse_pkginfo
  File "/app/env/lib/python3.4/site-packages/setuptools_scm-1.5.2-py3.4.egg/setuptools_scm/utils.py", line 64, in data_from_mime
  File "/usr/lib/python3.4/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc2 in position 8768: ordinal not in range(128)
```

This causes the version inference to fail and all versions of the irc package to install as 0.0.0.

---
- Bitbucket: https://bitbucket.org/pypa/setuptools_scm/issue/28
- Originally reported by: Jason R. Coombs
- Originally created at: 2015-06-11T04:12:40.903

**Comment by jaraco** [comment]
Indeed, in the environment where this is failing, LANG is not set.

---

Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
Setuptools forces UTF-8 for the encoding of package metadata. Use it also for reading. Fixes #28.

→ <<cset e8e83b611dc8>>

---

Original comment by: Jason R. Coombs
